modules/ROOT/pages/raise-error-component-reference.adoc

```diff
@@ -4,23 +4,90 @@ include::_attributes.adoc[]
 
-For core runtime errors types, you must use the implicit namespace and identifier, you can only customize the error's description message. +
-For custom error types, declare a new namespace. The namespace of an error type should help you identify the origin of an error. +
+== Raise Error Configuration
+
+[%header,cols="1,1,2,2"]
+|===
+| Field | Value | Description | Example
+
+| Description | String or DataWeave expression | Specifies the error description. |
+`description="Error description message."`
+
+| Type | String | Specifies the type of the error. |
+`type="CUSTOM:CUSTOM_ERROR"`
+
+|===
+
+== Raise Core Runtime Errors Types
+For core runtime errors types, you must use the implicit namespace and identifier, you can only customize the error's description message. For example:
+
+[source,xml,linenums]
+----
+<raise-error type="MULE:CONNECTIVITY" description="Error description message"/>
+----
+
+== Raise Custom Errors Types
+For custom error types, declare a new namespace. The namespace of an error type should help you identify the origin of an error. For example:
+
+[source,xml,linenums]
+----
+<raise-error type="ORDER:INVALID_DATA" description="Email is invalid. Cannot complete transaction"/>
+----
+
 You cannot use existing namespaces from connectors or modules, since these have their defined namespaces.
-Once you declare a custom namespace by using it in a `raise-error`, you can use it in other
-`raise-error` components and custom types.
+Once you declare a custom namespace by using it in a `raise-error`, you can use it in other `raise-error` components and custom types.
 
-The following example flow produces an `ACCOUNT:INSUFFICIENT_FUNDS` error when a
-transfer amount surpasses the available balance, preventing the transfer from
-taking place. Additionally, it generates a dynamic description by providing an
-expression with failure details.
 
-.Example: XML Configuration of the Flow
+== Examples
+
+The following example of a driving lessons platform checks that the user is over 16 years old to drive. The flow raises a `PRECONDITIONS:INCORRECT_AGE` error when the user enters an age lower than 16 years.
+
+* The Choice router components adds conditional processing to the flow.
+* If the number value is lower than 16 in the request `http://localhost:8081/test?age=15`, the Raise Error component raises the error type `PRECONDITIONS:INCORRECT_AGE` and description `Minimum age of 16 required to drive`.
+* Then, a Logger component logs the error message `Minimum age required to drive is 16 years old` in the On Error Continue component flow whose type is `PRECONDITIONS:INCORRECT_AGE`.
+* If the number value is greater than 16 in the request `http://localhost:8081/test?age=17`, a Logger component returns the default message `User age above 16 years. Allowed to drive`.
+* If no age value parameter is passed in the request `http://localhost:8081/test`, a Logger component returns the message `The parameter age is missing`.
```

Comment:
Maybe we can explain that the <choice> component generates the error automatically here: 

If no `age` parameter is passed in the request `http://localhost:8081/test`, the `choice` component will automatically raise a `MULE:EXPRESSION` error when trying to evaluate the expression `#[message.attributes.queryParams.age &lt; 16]` because no `age` parameter was provided in the request, so its value is `null`. 
Then, an On-Error Continue component matches the type of the error `MULE:EXPRESSION` and executes the Logger component inside its scope. The Logger component returns the message `The parameter age is missing`.

Moreover, the error message could be updated to `The parameter age is missing or is invalid`, because the error is generated when the `age` value cannot be compared to a number, this would also happen if you make the following request: http://localhost:8081/test?age=abc